src/Api/DataTransformers/ClientOutputDataTransformer.php

```diff
@@ -53,7 +53,7 @@ public function transform($data, string $to, array $context = [])
         $output->setId($data->getId());
         $output->setName($data->getName());
         $output->setUrl($data->getUrl());
-        $output->setQuota($data->getQuota());
+        $output->setQuota((int)$data->getQuota());
```

Comment:
Did you check by creating a client with a huge quota through the api?

Reply:
I've been testing and data is lost any ways. For example, I have created a client with quota 9223372034539499520, and with or without the casting it returns the value 9223372034539500000, the only difference are the quotes. 

Also, I was not aware that when you set a quota from the UI, the value is converted and saved to bytes. So I should change that for the api too.

Reply:
Is the issue about losing data also present in the Web UI or is it just the API ? If it happens in both cases then just open a new issue so that it is solved.

About converting to bytes. What is the unit that the Web UI uses ? Ks? Mbs? Is always fixed or the user can change it? If the user can change it at will then we should just use Bytes in the API, make them work and be happy. If it is hardcoded to one unit then yes, let's stick to that in the API so that we don't get inconsistencies. Let's see what @elacunza thinks about this.

Reply:
Its just in the API.
It uses GBs, and it's always fixed. What's more, I think there is a bug because each time I edit and save the client's form, the form tries to convert the quota to Bytes (also when it's already converted), so eventually it will exceed the accepted value range.

Reply:
So make API consistent, file a issue for the form. ;)

Reply:
If the quota value given is larger than the max_safe_int (9007199254740991), json will round the value, so the value given and the one that it saves will not be consistent. Do I handle it to throw an Exception if the quota value is higher?

Reply:
Check that it is not greater than the MAX_SAFE_INT, if it is then return an error ("Out of bounds" or something like that). That limit seems reasonable for know (petabytes of quota).

Also, it must be clear the unit for that field in the API.

Reply:
If [comment in the code](https://github.com/elkarbackup/elkarbackup/blob/7d7459fd65e1e18b15f8243a520bf7a556bf7ba2/src/Form/Type/ClientType.php#L28) is right, the actual persisted value is in KBytes, not Bytes.

[When saving it is multiplied twice by 1024](https://github.com/elkarbackup/elkarbackup/blob/7d7459fd65e1e18b15f8243a520bf7a556bf7ba2/src/Form/Type/ClientType.php#L33) and [when presenting divided twice](https://github.com/elkarbackup/elkarbackup/blob/7d7459fd65e1e18b15f8243a520bf7a556bf7ba2/templates/default/fields.html.twig#L73).

Quite a mess but if the actual UI does not better either let's just be practical and don't lose more time with this.

In the API we return the quota in GBs and we also expect them in GB. For that max_safe_int in json is more than enough. "int" should suffix in your API classes, no need to send or receive the number as a string.

The problem might arise when multiplying the received value twice by 1024 to save it in the entity as an int, but in a 64bit system the max value for PHP is 9,223,372,036,854,775,807, so we probably will be safe unless we go well over 8,000,000,000,000 Gbyte, The UI doesn't seem to pay attention to this either, so that will be good enough for now.

To sum up:
- No fancy "string" number in the JSON just a regular int.
- Multiple and divide when going from Doctrine Entity <=> API Dto
- Make sure that for a reasonable high value (for example 123,456 Gbytes) it works as expected

This test will probably fail in a 32bit system, but that's on purpose, it seems the UI has the same issue and I'd rather tests fail and surface the problem when run in a 32bit system than let the user think that everything is ok.

Reply:
If there's no compelling reason to use GB in API, I'll prefer to use MiB, so that quotas smaller than 1GB can be configured.

Reply:
Ok, so then MiB for the API (receive and return) and save it in KB?

Reply:
Save it the same as does the WUI.

Reply:
Ok, I will make that changes in other branch as it is not a test related issue

Reply:
@elacunza keep in mind that if you use MB in the API and then view or update the client in the GUI you will probably lose that detail.

Reply:
Sorry, what is using WUI? I thought it was using MiB? (because of 1024*1024 multiplication...)

Reply:
Changes made in PR #560

Reply:
@uanabitarte PR #560 is merged, merge it in this branch.
Do tests still need to be changed or they already work as commented?
Once this is confirmed I can resolve this conversation and Eneko will be able to merge it.

Reply:
I think the only issue left is that the tests do not check if there is a well known client or job in the collection (GET /clients or /jobs). I could not find a way to assert the content of the array inside the json.
I.e., the response is a json and the elements are in the 'hydra:member' field, that is an array of jsons, so it should check whether one of those json matches with the well known element. 
There is a assertArraySubset function, but as I have read, it only works for associative arrays.

Reply:
I think you should be able to convert to PHP object/array any JSON string with json_decode() . You can then check array's values with in_array for example.

Reply:
@uanabitarte checking that the result holds the expected data is important, so please try @elacunza's suggested approach.

If I'm not getting this wrong this is not related with the quota issue but I'll just unresolve this conversation to not generate more noise, once solved pleas resolve the conversation and I will approve.